src/components/Input/index.tsx

```diff
@@ -38,20 +44,27 @@ const Input: React.FC<Props> = props => {
             <RenderIf isTrue={!!label}>
                 <Label>{label}</Label>
             </RenderIf>
-            <StyledInput
-                onChangeText={onChange}
-                value={value}
-                placeholder={placeholder}
-                editable={isEnabled}
-                selectTextOnFocus={isEnabled}
-                error={error}
-                isFocused={isFocused}
-                onBlur={() => setFocusState(false)}
-                onFocus={() => setFocusState(true)}
-                disabled={disabled}
-                keyboardType={keyboardType}
-                autoFocus={autoFocus}
-            />
+            <View>
+                <StyledInput
+                    onChangeText={onChange}
+                    value={value}
+                    placeholder={placeholder}
+                    editable={isEnabled}
+                    selectTextOnFocus={isEnabled}
+                    error={error}
+                    isFocused={isFocused}
+                    onBlur={() => setFocusState(false)}
+                    onFocus={() => setFocusState(true)}
+                    disabled={disabled}
+                    keyboardType={keyboardType}
+                    autoFocus={autoFocus}
+                    icon={!!icon}
```

Comment:
Since what is passed to StyledInput is not the icon itself but a boolean representing that there is an icon passed, we should name it hasIcon like:
 ```hasIcon={!!icon}```

Reply:
dope!

Reply:
in the next line I am actually passing the icon prop 
```
<RenderIf isTrue={!!icon}>
      <Icon iconPosition={iconPosition}>{icon}</Icon>
</RenderIf>
```